lambdas/drivingpermitcheck/src/main/java/uk/gov/di/ipv/cri/drivingpermit/api/handler/DrivingPermitHandler.java

```diff
@@ -320,9 +322,9 @@ private IdentityVerificationService createIdentityVerificationService(
                 new FormDataValidator(), null, serviceFactory.getEventProbe());
     }
 
-    private ThirdPartyAPIService selectThirdPartyAPIService() {
+    private ThirdPartyAPIService selectThirdPartyAPIService(String documentCheckingRouteHeader) {
         // TODO Change to a switch for each api and read feature flag + header
-        if (configurationService.getUseLegacy()) {
+        if (configurationService.getUseLegacy() && "dcs".equals(documentCheckingRouteHeader)) {
```

Comment:
Have we tested this/ran the ac tests against this on a branch, I'm worried about what happens if "dcs" is not passed in the header but we still want to route the user to dcs. I don't see where dcs is being set in the header if it's supposed to be and I worry that without it users will default to the dvaThirdPartyApiService

Reply:
I think what we want to do here is just flip this statement. We currently have useLegacy this logic becomes easier if we make that like passport and have useDvaDirect. Then
```
    private ThirdPartyAPIService selectThirdPartyAPIService(
            boolean dvaDirectEnabled, boolean newThirdpartyAPI) {
        // Feature flag and header required for new api
        if (dvaDirectEnabled && newThirdpartyAPI) {
            // DVAD
            return thirdPartyAPIServiceFactory.getDvaThirdPartyService();
        } else {
            // Legacy DCS
            return thirdPartyAPIServiceFactory.getDcsThirdPartyAPIService();
        }
    }
```

Reply:
Will make that correction now.